### DISCUSSION TITLE
prometheus ipv6 Targets connect: invalid argument

### DISCUSSION THREAD

**Discussion by lvchongyi** [prometheus/prometheus#9636] [Q&A]
environment：
  CentOS Linux release 7.2.1511
  prometheus-2.29.1 
  IPv6

config：
prometheus.yml
  - job_name: IPv6
    metrics_path: /metrics
    honor_labels: true
    static_configs:
      - targets: ["[fe80::5054:ff:fe16:caa8]:9999"]
        labels:
          compType: IPv6
          clusterName: newland
          svcId: 1
          svcUnitId: 11

error:
Get "http://[fe80::5054:ff:fe16:caa8]:9999/metrics": dial tcp [fe80::5054:ff:fe16:caa8]:9999: connect: invalid argument

**Comment by roidelapluie** [comment]
if you want to monitor with link-local addresses, the address should contain the interface, e.g. "[fe80::5054:ff:fe16:caa8%eth0]:9999"

This is not a prometheus issue, but an IPv6 thing.

**Reply by lvchongyi** [reply]
prometheus.yml
 - job_name: IPv6
    metrics_path: /metrics
    honor_labels: true
    static_configs:
      - targets: ["[fe80::5054:ff:fe16:caa8%eth0]:9999"]
        labels:
          compType: IPv6
          clusterName: newland
          svcId: 1
          svcUnitId: 11

 add interface ,UI error:
Error: Failed to construct 'URL': Invalid URL

**Reply by roidelapluie** [reply]
Is it scraping correctly, besides the UI error?

**Reply by lvchongyi** [reply]
scrape has problem also，
execute result is:   Empty query result

**Reply by roidelapluie** [reply]
Can you run Prometheus with --log.level=debug and tell us the scrape error message?

**Comment by lvchongyi** [comment]
[root@vcapp27 ~]# /usr/lib/prometheus/prometheus  --log.level=debug --config.file=/usr/lib/prometheus/prometheus.yml
level=info ts=2021-11-03T02:51:29.796Z caller=main.go:400 msg="No time or size retention was set so using the default time retention" duration=15d
level=info ts=2021-11-03T02:51:29.796Z caller=main.go:438 msg="Starting Prometheus" version="(version=2.30.3, branch=HEAD, revision=f29caccc42557f6a8ec30ea9b3c8c089391bd5df)"
level=info ts=2021-11-03T02:51:29.796Z caller=main.go:443 build_context="(go=go1.17.1, user=root@5cff4265f0e3, date=20211005-16:10:52)"
level=info ts=2021-11-03T02:51:29.796Z caller=main.go:444 host_details="(Linux 3.10.0-327.el7.x86_64 #1 SMP Thu Nov 19 22:10:57 UTC 2015 x86_64 vcapp27 (none))"
level=info ts=2021-11-03T02:51:29.796Z caller=main.go:445 fd_limits="(soft=65536, hard=65536)"
level=info ts=2021-11-03T02:51:29.796Z caller=main.go:446 vm_limits="(soft=unlimited, hard=unlimited)"
level=info ts=2021-11-03T02:51:29.800Z caller=web.go:541 component=web msg="Start listening for connections" address=0.0.0.0:9090
level=info ts=2021-11-03T02:51:29.800Z caller=main.go:822 msg="Starting TSDB ..."
level=info ts=2021-11-03T02:51:29.800Z caller=repair.go:57 component=tsdb msg="Found healthy block" mint=1635876001894 maxt=1635883200000 ulid=01FKH77B7B547J4M9D4GQW03DA
level=info ts=2021-11-03T02:51:29.801Z caller=repair.go:57 component=tsdb msg="Found healthy block" mint=1635883201894 maxt=1635890400000 ulid=01FKHE307J82SPCM4BA0HZEZ7R
level=info ts=2021-11-03T02:51:29.801Z caller=repair.go:57 component=tsdb msg="Found healthy block" mint=1635833299200 maxt=1635876000000 ulid=01FKHE30CHJGBW5S4EB0FJ4JGG
level=info ts=2021-11-03T02:51:29.801Z caller=repair.go:57 component=tsdb msg="Found healthy block" mint=1635890401894 maxt=1635897600000 ulid=01FKHMYSQB5CT7K42WGX16R6GE
level=info ts=2021-11-03T02:51:29.801Z caller=tls_config.go:191 component=web msg="TLS is disabled." http2=false
level=info ts=2021-11-03T02:51:29.806Z caller=head.go:479 component=tsdb msg="Replaying on-disk memory mappable chunks if any"
level=info ts=2021-11-03T02:51:29.809Z caller=head.go:513 component=tsdb msg="On-disk memory mappable chunks replay completed" duration=3.784628ms
level=info ts=2021-11-03T02:51:29.809Z caller=head.go:519 component=tsdb msg="Replaying WAL, this may take a while"
level=info ts=2021-11-03T02:51:29.814Z caller=head.go:555 component=tsdb msg="WAL checkpoint loaded"
level=info ts=2021-11-03T02:51:29.873Z caller=head.go:590 component=tsdb msg="WAL segment loaded" segment=20 maxSegment=26
level=info ts=2021-11-03T02:51:29.937Z caller=head.go:590 component=tsdb msg="WAL segment loaded" segment=21 maxSegment=26
level=info ts=2021-11-03T02:51:29.945Z caller=head.go:590 component=tsdb msg="WAL segment loaded" segment=22 maxSegment=26
level=info ts=2021-11-03T02:51:29.969Z caller=head.go:590 component=tsdb msg="WAL segment loaded" segment=23 maxSegment=26
level=info ts=2021-11-03T02:51:29.970Z caller=head.go:590 component=tsdb msg="WAL segment loaded" segment=24 maxSegment=26
level=info ts=2021-11-03T02:51:29.973Z caller=head.go:590 component=tsdb msg="WAL segment loaded" segment=25 maxSegment=26
level=info ts=2021-11-03T02:51:29.973Z caller=head.go:590 component=tsdb msg="WAL segment loaded" segment=26 maxSegment=26
level=info ts=2021-11-03T02:51:29.973Z caller=head.go:596 component=tsdb msg="WAL replay completed" checkpoint_replay_duration=4.519105ms wal_replay_duration=159.092582ms total_replay_duration=167.413774ms
level=info ts=2021-11-03T02:51:29.975Z caller=main.go:849 fs_type=XFS_SUPER_MAGIC
level=info ts=2021-11-03T02:51:29.975Z caller=main.go:852 msg="TSDB started"
level=debug ts=2021-11-03T02:51:29.975Z caller=main.go:853 msg="TSDB options" MinBlockDuration=2h MaxBlockDuration=1d12h MaxBytes=0B NoLockfile=false RetentionDuration=15d WALSegmentSize=0B AllowOverlappingBlocks=false WALCompression=true
level=info ts=2021-11-03T02:51:29.975Z caller=main.go:979 msg="Loading configuration file" filename=/usr/lib/prometheus/prometheus.yml
level=debug ts=2021-11-03T02:51:29.976Z caller=manager.go:195 component="discovery manager scrape" msg="Starting provider" provider=static/0 subs=[prometheus]
level=debug ts=2021-11-03T02:51:29.976Z caller=manager.go:195 component="discovery manager scrape" msg="Starting provider" provider=static/1 subs=[IPv6]
level=debug ts=2021-11-03T02:51:29.976Z caller=manager.go:213 component="discovery manager scrape" msg="Discoverer channel closed" provider=static/0
level=debug ts=2021-11-03T02:51:29.976Z caller=manager.go:195 component="discovery manager notify" msg="Starting provider" provider=static/0 subs=[config-0]
level=info ts=2021-11-03T02:51:29.976Z caller=main.go:1016 msg="Completed loading of configuration file" filename=/usr/lib/prometheus/prometheus.yml totalDuration=848.718µs db_storage=1.08µs remote_storage=4.285µs web_handler=1.99µs query_engine=1.059µs scrape=260.715µs scrape_sd=117.698µs notify=30.983µs notify_sd=22.58µs rules=3.632µs
level=info ts=2021-11-03T02:51:29.976Z caller=main.go:794 msg="Server is ready to receive web requests."
level=debug ts=2021-11-03T02:51:29.976Z caller=manager.go:213 component="discovery manager scrape" msg="Discoverer channel closed" provider=static/1
level=debug ts=2021-11-03T02:51:29.976Z caller=manager.go:213 component="discovery manager notify" msg="Discoverer channel closed" provider=static/0
